src/slider.jsx

```diff
@@ -323,17 +314,84 @@ const Slider = React.createClass({
       document.addEventListener('touchcancel', this._dragTouchEndHandler, false);
     }
     this._onDragStart(e);
+
+    // Cancel scroll and context menu
+    e.preventDefault();
   },
 
   _onHandleMouseDown(e) {
     if (document) {
       document.addEventListener('mousemove', this._dragHandler, false);
       document.addEventListener('mouseup', this._dragEndHandler, false);
-      this._toggleSelection('none');
+
+      // Cancel text selection
+      e.preventDefault();
+
+      // Set focus manually since we called preventDefault()
+      ReactDOM.findDOMNode(this.refs.handle).focus();
```

Comment:
Another comment: for a long time now, I have seen the following phrase in the [React docs](https://facebook.github.io/react/docs/more-about-refs.html):

> React also supports using a string (instead of a callback) as a ref prop on any component, although this approach is mostly legacy at this point.

It seems the right approach would be use the callback, but this phrase is there for a long time.